tensordict/nn/distributions/composite.py

```diff
@@ -147,19 +152,67 @@ def rsample(self, shape=None) -> TensorDictBase:
             shape + self.batch_shape,
         )
 
-    def log_prob(self, sample: TensorDictBase) -> TensorDictBase:
-        """Writes a ``<sample>_log_prob entry`` for each sample in the input tensordict, along with a ``"sample_log_prob"`` entry with the summed log-prob."""
+    def log_prob(self, sample: TensorDictBase) -> torch.Tensor:
```

Comment:
This is BC-breaking.
I would suggest the following:
- add a `aggregate_probabilities` argument in the constructor, default it to `None`
- if `None`, raise a warning saying that as of now (v0.5) we will default it to `False` but as of v0.7 it will be `True`. People can remove the warning by passing this value explicitly.
- if `False`, `log_prob` calls `log_prob_composite`. If `True`, it behaves like your implementation.
- The same argument controls the behaviour of `log_prob` and `entropy`.